### ISSUE TITLE
CPython3 - 'System.Reflection.Emit.TypeBuilder' was not found.

### ISSUE THREAD

**Issue by Cyril-Pop** [DynamoDS/DynamoRevit#3008]
### Dynamo Version

Dynamo 2.19.3

### Revit Version

Revit 2024.2

### Operating System

Windows 11

### What did you do?

try to use events subscription with CPython3 

### What did you expect to see?

No Error

### What did you see instead?

a Python error
```
Node Name: PythonScript
Package: Core.Scripting
Dynamo Version: 2.19.3.6394
Host: Dynamo Revit
Messages: PythonEvaluator.Evaluate operation failed.
The constructor on type 'System.Reflection.Emit.TypeBuilder' was not found.
Status: Warning
```
![image](https://github.com/DynamoDS/DynamoRevit/assets/53660624/1878fac8-906f-4e85-b521-516bdb669dd3)

Example 1

```
import clr
import System
from System import EventHandler

clr.AddReference("System.Drawing")
clr.AddReference("System.Windows.Forms")
import System.Drawing
import System.Windows.Forms

from System.Drawing import *
from System.Windows.Forms import *

class Form26(Form):
	def __init__(self):
		self.InitializeComponent()
	
	def InitializeComponent(self):
		self._button1 = System.Windows.Forms.Button()
		self.SuspendLayout()
		# 
		self._button1.Location = System.Drawing.Point(73, 103)
		self._button1.Name = "button1"
		self._button1.Size = System.Drawing.Size(142, 43)
		self._button1.TabIndex = 0
		self._button1.Text = "button1"
		self._button1.UseVisualStyleBackColor = True
		self._button1.Click += self.Button1Click
		# 
		self.ClientSize = System.Drawing.Size(284, 261)
		self.Controls.Add(self._button1)
		self.Name = "Form26"
		self.Text = "Form26"
		self.ResumeLayout(False)

	def Button1Click(self, sender, e):
		pass
  
objForm = Form26()
objForm.ShowDialog()
```

Example 2

```
import clr
import System
from System import EventHandler

clr.AddReference("RevitServices")
import RevitServices
from RevitServices.Persistence import DocumentManager
from RevitServices.Transactions import TransactionManager

# Import RevitAPI
clr.AddReference("RevitAPI")
import Autodesk
from Autodesk.Revit.DB import *

clr.AddReference("RevitAPIUI")
from Autodesk.Revit.UI import *
from Autodesk.Revit.UI.Events import *

doc = DocumentManager.Instance.CurrentDBDocument
uidoc=DocumentManager.Instance.CurrentUIApplication.ActiveUIDocument
uiapp = uidoc.Application

def dismiss (sender, eventArgs):
    if isinstance(eventArgs, TaskDialogShowingEventArgs):
        eventArgs.OverrideResult(TaskDialogResult.No)
    return

dialogEventHandler = EventHandler[DialogBoxShowingEventArgs](dismiss)

uiapp.DialogBoxShowing += dialogEventHandler
```

### Details

the error seems related to the build/compilation of PythonNet 2.5.2 (Python.Runtime) with Net.Standard 2.0 as target

there are potentially others PythonNet 2.5.2 issues due to this change (Net Framework -> Net.Standard 2.0), 
_will probably be resolved in PythonNet 3.x.x_

**Comment by Cyril-Pop** [comment]
Related issue
https://github.com/DynamoDS/Dynamo/issues/14612

**Comment by AhmadEltobshy** [comment]
no updates ? 

i have faced the same issue 
![image](https://github.com/DynamoDS/DynamoRevit/assets/74884132/5a5bf30a-c7fc-4898-be0d-63646c5bfbfa)

**Comment by Amoursol** [comment]
@Cyril-Pop @AhmadEltobshy We have plans to update the PythonNET bridge project as per our [roadmap](https://dynamobim.org/roadmap/) that should resolve a lot of CPython parity issues with IronPython. We're not yet sure of the full extent, but it should be significantly better than today.

**Comment by Cyril-Pop** [comment]
@Amoursol 
Do you have an approximate date for the integration of Pythonnet 3?

**Comment by pak9819** [comment]
I have the same error when using winforms.

**Comment by Narimannp** [comment]
Was anybody able to find a solution for that? It seems that this is due to the fact that dynamo 2.19 uses cpython and not ironpython anymore

**Comment by qyalexwen** [comment]
This issue is back after Revit update 2024.3 is installed.

**Comment by Narimannp** [comment]
You can simply switch the python node to iron python 2

**Comment by ChrisG-84** [comment]
I've tried editing the AppData DynamoSettings.xml to add a custom Python engine and that does not work, 

I have tried for days to force Dynamo Python to import a different Python.Runtime.dll that is newer than 2.5, as well as import python packages from a different source folder than the embedded python location. Sometimes it works like for Numpy etc. but clr-loader & pythonnet do not work the right way.

Python.Runtime.dll version: 2.5.2.7195,
Python.Runtime.dll location: C:\Program Files\Autodesk\Revit 2024\AddIns\DynamoForRevit\Python.Runtime.dll,
Python Script: CLR imported successfully
Python Script: Loading Windows Forms assemblies...
Python Script: Failed during .NET integration test: module 'clr' has no attribute '_add_pending_namespaces'
Python Script: Traceback: Traceback (most recent call last):
I did a bunch of things and cannot recall how this error started to occur but I'm really trying to use PythonNet3.01

While loading assembly Python.Runtime, Version=3.0.1.0, Culture=neutral, PublicKeyToken=5000fea6cba702dd, Dynamo detected that the dependency Python.Runtime, Version=3.0.1.0, Culture=neutral, PublicKeyToken=5000fea6cba702dd was already loaded with an incompatible version.
